test/unit/spec/common/jobManager-Spec.js

```diff
@@ -769,13 +770,64 @@ define([
         });
 
         /* defaultHandlerMixin */
-        function setUpHandlerTest(context, event) {
+        function setUpHandlerTest(context, event, channelType, channelIdList) {
             context.jobManagerInstance.addEventHandler(event, { handler_1: scream });
             expect(Object.keys(context.jobManagerInstance.handlers[event]).sort()).toEqual([
                 `__default_${event}`,
                 'handler_1',
             ]);
-            context.jobManagerInstance.addListener(event, jcm.CHANNEL.JOB, [context.jobId]);
+            context.jobManagerInstance.addListener(event, channelType, channelIdList);
+        }
+
+        function getMessages(msgType) {
+            const error = 'some error string';
+            // all the data
+            let dataSource = {},
+                // the batch ID for the data source
+                batchId;
+            if (msgType === jcm.MESSAGE_TYPE.INFO) {
+                // filter out the non-batch jobs
+                Object.values(JobsData.example.INFO.valid).forEach((jobInfo) => {
+                    if ('batch_id' in jobInfo && jobInfo.batch_id) {
+                        dataSource[jobInfo.job_id] = jobInfo;
+                    }
+                });
+                batchId = Object.values(dataSource)[0].batch_id;
+            } else if (msgType === jcm.MESSAGE_TYPE.RETRY) {
+                dataSource = JobsData.batchJob.retryMessages;
+                batchId = JobsData.batchJob.batchId;
+            }
+
+            // use a job in the batch, not the batch parent
+            const jobIdList = Object.keys(dataSource),
+                jobId = jobIdList[0] === batchId ? jobIdList[1] : jobIdList[0];
+
+            return {
+                'by job ID': {
```

Comment:
for the default handler tests, I wanted to test the handlers would work correctly with different types of message:
- a standard message using the JOB channel and addressed by job ID;
- a message containing data for a single job within a batch, using the BATCH channel and addressed by batch ID;
- as the previous, but containing data for numerous jobs